user/what-is-a-workspace.rst

```diff
@@ -0,0 +1,314 @@
+
+.. code-block:: python
+
+	import foo
+
+	def test_foo():
+	    assert foo.foo_func()
+
+
+Tell ``colcon`` to run the tests.
+
+.. code-block:: bash
+
+	# Make sure you run this command from the root of the workspace!
+	colcon test
+
+Lets look in the ``log`` directory again.
+
+::
+
+	log
+	├── build_2022-05-20_11-50-03/...
+	├── COLCON_IGNORE
+	├── latest -> latest_test
+	├── latest_build -> build_2022-05-20_11-50-03
+	├── latest_test -> test_2022-05-20_11-50-05
+	└── test_2022-05-20_11-50-05
+	    ├── events.log
+	    ├── foo
+	    │	├── command.log
+	    │	├── stderr.log
+	    │	├── stdout.log
+	    │	├── stdout_stderr.log
+	    │	└── streams.log
+	    └── logger_all.log
+
+
+A new symlink ``latest_test`` was created, and it points to a new directory ``log/test_<date and time>``.
+This holds the console output from running the test.
+Checkout ``stdout_stderr.log``  and see the output of the latest test
+
+.. code-block:: bash
+
+	cat log/latest_test/foo/stdout_stderr.log
+
+.. note::
+
+	The test's output can be printed to the console in real time with
+	``colcon test --event-handlers console_direct+``.
+
+.. note::
+
+	Use the command ``colcon test-result`` to see a summary of test results on the console after tests have been run.
+
+
+
+Install artifacts
+*****************
+
+The last directory to talk about is the ``install`` directory.
+It contains both the installed software, and shell scripts that enable you to use it.
+This is sometimes called the **install space**.
+
+.. note::
+
+	You can change where packages are installed to with the ``--install-base`` option to ``colcon build``.
+
+Let's look inside.
+
+::
+
+	install
+	├── COLCON_IGNORE
+	├── foo/...
+	├── local_setup.[bash|bat|ps1|sh|zsh|...]
+	├── _local_setup_util_[sh|ps1|...].py
+	└── setup.[bash|bat|ps1|sh|zsh|...]
+
+The package ``foo`` was installed into the directory ``install/foo``.
+By default colcon builds an **isolated workspace**.
+That means every package is installed into its own directory.
+
+The shell scripts set environment variables that allow you to use the the software.
+You'll need to invoke the shell scripts, which is commonly called **sourcing a workspace**.
+
+.. note::
+
+	Always source a workspace from a different terminal than the one you used ``colcon build``.
+	Failure to do so can prevent colcon from detecting incorrect dependencies.
+
+..
+	TODO(sloretz) what's the difference between local_setup and setup?
+
+Source the workspace using the appropriate script for your shell.
+
+``sh`` compatible shells:
+
+.. code-block:: sh
+
+	# Note the . at the front; that's important!
+	. install/setup.sh
+
+``bash``:
+
+.. code-block:: bash
+
+	source install/setup.bash
+
+Windows ``cmd.exe``:
+
+.. code-block:: bat
+
+	call install/setup.bat
+
+
+.. note::
+
+	Always source a workspace from a different terminal than the one you used ``colcon build``.
+	Failure to do so can prevent colcon from detecting incorrect dependencies.
```

Comment:
Unintentional. I moved the notes around a few times while writing this and must have copied instead of cut one of the times. Fixed in 4e92dcdf43ec9c4671c1a9612becdd0d5844c955